src/ralph/conf.py

```diff
@@ -0,0 +1,235 @@
+"""Configurations for Ralph"""
+
+import io
+from os import environ
+from pathlib import Path
+from typing import Literal, Union
+
+from click import get_app_dir
+from pydantic import BaseModel, BaseSettings
+
+from ralph.utils import import_string
+
+APP_DIR = Path(environ.get("RALPH_APP_DIR", get_app_dir("ralph")))
```

Comment:
Can we define it in the main Settings class and self reference this field in the rest of the configuration?

Reply:
I don't find an easy way to achieve this.

Tried something like this:
```
PATH: str = Field(
    default_factory=lambda: str(Settings.__fields__["APP_DIR"].default / "archives")
)
```
However, this approach only uses the static default value and doesn't get updated if the default is changed via an environment variable (which happens at instantiation) and it doesn't work in the nested Config class :disappointed:.

Another solution might be defining a second settings class (something like `CoreSettings`), which defines APP_DIR and is instantiated prior to defining Settings. WDYT?

Reply:
The following snippet just works :tm::

```python
from os import environ
from pathlib import Path

from pydantic import BaseSettings


class Settings(BaseSettings):

    ROOT_DIR: Path = Path(environ.get("HOME", Path.cwd())) / ".config/myapp"
    HISTORY_FILE_PATH: Path = ROOT_DIR / "history.json"


settings = Settings()
print(settings)
```

The output:

```
ROOT_DIR=PosixPath('/home/maupetit/.config/myapp')
HISTORY_FILE_PATH=PosixPath('/home/maupetit/.config/myapp/history.json')                                                                                                                                                                               
```

WDYT?

Reply:
That's Awesome! I wasn't aware of this) Thanks) However, I think two issues remain: 

We use the `APP_DIR` in the nested Config metaclass:

```python
class Settings(BaseSettings):
    class Config:
        env_file = APP_DIR / ".env"
```

and in `FSStorageBackendSettings`, a separate model that is inside Settings (has to be defined before).

Reply:
Ok, then. I think your first idea to define separated core settings solves all issues you raised. Go!

```python
from os import environ
from pathlib import Path

from pydantic import BaseSettings


class CoreSettings(BaseSettings):

    ROOT_DIR: Path = Path(environ.get("HOME", Path.cwd())) / ".config/myapp"


class Settings(BaseSettings):

    _CORE: BaseSettings = CoreSettings()
    HISTORY_FILE_PATH: Path = _CORE.ROOT_DIR / "history.json"

    class Config:
        env_file: str = CoreSettings().ROOT_DIR / ".env"


settings = Settings()
print(settings)
```

Reply:
Thank you for your help :pray: Updated this part)